### PR TITLE
Updates in flutter/flow for the shell refactor (Patch 4).

### PR DESCRIPTION
**Pull request by chinmaygarde**
This changes is part of a large patch for easier review. Try the whole patch in one go by checking out https://github.com/chinmaygarde/flutter_engine/tree/shell directly. This patch contains the following changes:

* Allows multiple compositor context implementations which provides the necessary hooks for fuchsia to priovide its own rasterizer backend. Refer to Patch 3 for use.
* Remove access to task runners via the singleton accessors.
* Remove memory statistics performace overlays that were provided by the engine but never enabled by the framework and only hooked up on Darwin platforms.
* Remove most of the Fuchsia specific define guards as these have been worked around in the compositor on that platform (again, refer to Patch 3).

### CODE REVIEW

File: flow/compositor_context.h [hunk 1]
```diff
@@ -18,72 +17,65 @@
...
-  TextureRegistry& texture_registry() { return *texture_registry_; }
+  TextureRegistry& texture_registry() { return texture_registry_; }
 
   const Counter& frame_count() const { return frame_count_; }
 
   const Stopwatch& frame_time() const { return frame_time_; }
 
   Stopwatch& engine_time() { return engine_time_; }
 
-  const CounterValues& memory_usage() const { return memory_usage_; }
```
Comment (cbracken) [inline flow/compositor_context.h]: Was wondering where this went and the answer is apparently we never used it. Nice! :)

Comment (chinmaygarde) [reply flow/compositor_context.h]: This was only ever wired up on iOS and never exposed in the framework.

File: flow/skia_gpu_object.h [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  ~SkiaUnrefQueue();
+
+  FRIEND_REF_COUNTED_THREAD_SAFE(SkiaUnrefQueue);
+  FRIEND_MAKE_REF_COUNTED(SkiaUnrefQueue);
+  FXL_DISALLOW_COPY_AND_ASSIGN(SkiaUnrefQueue);
+};
+
+/// An object whose deallocation needs to be performed on an specific unref
+/// queue. The template argument U need to have a call operator that returns
```
Comment (cbracken) [inline flow/skia_gpu_object.h]: nit: s/need/needs/

**Comment by cbracken** [comment]
lgtm

**Comment by chinmaygarde** [comment]
Landed in https://github.com/flutter/engine/pull/4932.

**Pull request closed without merging**